The easiest way to illustrate the current coverage is this:
I've added `raise ValueError(int)` to all four methods.
<img width="752" alt="Снимок экрана 2022-01-18 в 10 11 40" src="https://user-images.githubusercontent.com/4660275/149888281-908e16d5-bb0e-4d6a-a15a-c53936477399.png">

And here's the result of both files on `main` branch:

```
» ./python.exe -m test -v test_typing -v test_types
======================================================================
ERROR: test_or_type_operator_with_Alias (test.test_types.UnionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython/Lib/test/test_types.py", line 849, in test_or_type_operator_with_Alias
    assert typing.List | str == typing.Union[typing.List, str]
           ~~~~~~~~~~~~^~~~~
  File "/Users/sobolev/Desktop/cpython/Lib/typing.py", line 1152, in __or__
    raise ValueError(3)
    ^^^^^^^^^^^^^^^^^^^
ValueError: 3

======================================================================
ERROR: test_or_type_operator_with_SpecialForm (test.test_types.UnionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython/Lib/test/test_types.py", line 872, in test_or_type_operator_with_SpecialForm
    assert typing.Optional[int] | str == typing.Union[typing.Optional[int], str]
           ~~~~~~~~~~~~~~~~~~~~~^~~~~
  File "/Users/sobolev/Desktop/cpython/Lib/typing.py", line 1029, in __or__
    raise ValueError(1)
    ^^^^^^^^^^^^^^^^^^^
ValueError: 1

======================================================================
ERROR: test_or_types_operator (test.test_types.UnionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython/Lib/test/test_types.py", line 641, in test_or_types_operator
    self.assertEqual(typing.List | typing.Tuple, typing.Union[typing.List, typing.Tuple])
                     ~~~~~~~~~~~~^~~~~~~~~~~~~~
  File "/Users/sobolev/Desktop/cpython/Lib/typing.py", line 1152, in __or__
    raise ValueError(3)
    ^^^^^^^^^^^^^^^^^^^
ValueError: 3

======================================================================
ERROR: test_union_args (test.test_types.UnionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython/Lib/test/test_types.py", line 749, in test_union_args
    check(typing.Union[int, str] | list, (int, str, list))
          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
  File "/Users/sobolev/Desktop/cpython/Lib/typing.py", line 1029, in __or__
    raise ValueError(1)
    ^^^^^^^^^^^^^^^^^^^
ValueError: 1

======================================================================
ERROR: test_union_parameter_substitution (test.test_types.UnionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython/Lib/test/test_types.py", line 788, in test_union_parameter_substitution
    x = int | T | bytes
        ~~~~~~~~^~~~~~~
  File "/Users/sobolev/Desktop/cpython/Lib/typing.py", line 1029, in __or__
    raise ValueError(1)
    ^^^^^^^^^^^^^^^^^^^
ValueError: 1

----------------------------------------------------------------------
Ran 108 tests in 0.119s

FAILED (errors=5, skipped=1)
test test_types failed
test_types failed (5 errors)

== Tests result: FAILURE ==

1 test OK.

1 test failed:
    test_types

Total duration: 885 ms
Tests result: FAILURE
```

We expect all four to be raised, but got only `1` and `3`.
So, even with `test_types`, not everything is covered.

Thanks for letting me know! 👍

--- a/Lib/test/test_typing.py
+++ b/Lib/test/test_typing.py
@@ -2468,16 +2468,6 @@ class GenericTests(BaseTestCase):
                     class Foo(obj):
                         pass
 
-    def test_or_and_ror(self):
-        self.assertEqual(
-            SimpleMapping[str, int] | bool,
-            Union[SimpleMapping[str, int], bool],
-        )
-        self.assertEqual(
-            bool | SimpleMapping[str, int],
-            Union[bool, SimpleMapping[str, int]],
-        )
-
 
 class ClassVarTests(BaseTestCase):
 

--- a/Lib/test/test_typing.py
+++ b/Lib/test/test_typing.py
@@ -3926,8 +3926,8 @@ class CollectionsAbcTests(BaseTestCase):
         self.assertIsSubclass(B, typing.Mapping)
 
     def test_or_and_ror(self):
-        self.assertEqual(typing.Sized | int, Union[typing.Sized, int])
-        self.assertEqual(int | typing.Hashable, Union[int, typing.Hashable])
+        self.assertEqual(typing.Sized | typing.Awaitable, Union[typing.Sized, typing.Awaitable])
+        self.assertEqual(typing.Coroutine | typing.Hashable, Union[typing.Coroutine, typing.Hashable])
 
 
 class OtherABCTests(BaseTestCase):

--- a/Lib/test/test_typing.py
+++ b/Lib/test/test_typing.py
@@ -529,8 +529,8 @@ class BaseCallableTests:
 
     def test_or_and_ror(self):
         Callable = self.Callable
-        self.assertEqual(Callable | bool, Union[Callable, bool])
-        self.assertEqual(bool | Callable, Union[bool, Callable])
+        self.assertEqual(Callable | Tuple, Union[Callable, Tuple])
+        self.assertEqual(Tuple | Callable, Union[Tuple, Callable])
 
     def test_basic(self):
         Callable = self.Callable

